### DISCUSSION TITLE
Welcome to cypress-allure-plugin Discussions!

### DISCUSSION THREAD

**Discussion by Shelex** [Shelex/cypress-allure-plugin#56] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

**Comment by jamilhf** [comment]
hi are there any integrations with jenkins , if yes can you share the configuration please,

thanks

**Reply by Shelex** [reply]
> hi are there any integrations with jenkins , if yes can you share the configuration please,
> 
> thanks

Hi.
Actually, integration with CI is not a part of this plugin, as it is just an interface to prepare data-files to generate Allure report later.
However I know there is a plugin for jenkins: [link](https://plugins.jenkins.io/allure-jenkins-plugin/), maybe it would help.

**Comment by SaadUsmanitk** [comment]
hi, i have installed the allure plugin and reports are now showing offline but i wanted to know how to configure on github? My pipeline is configured for test cases to run but iam confused how pages are generated on github? how index.html file is geneated,
 any step by step explanation available that would be helpful. Iam just a beginner both for git and cypress

**Reply by Shelex** [reply]
Hi!
 - for github pages information is available here: https://pages.github.com/
 - there is pipeline example in [cypress-allure-plugin-example](https://github.com/Shelex/cypress-allure-plugin-example/blob/master/.github/workflows/allure.yaml). Basically, you just clear your folders and prepare history data, run tests, generate report with some github action ( i have used [QuintilianoNery/action-allure-report-V2@0.2.0](https://github.com/marketplace/actions/allure-report-v2)) and do commit and push with generated report to your main branch - that is it.

In general, if you are a beginner I would recommend to avoid any guides but follow documentation.
 
 > how to configure on github

With yaml file in `.github/workflows` folder, they have great [quickstart doc](https://docs.github.com/en/actions/quickstart)

> My pipeline is configured for test cases to run

That is great! all you need to do - just add some steps that handle report generation and commit+push.
 
 > how pages are generated on github

Github does not generate any pages
 > how index.html file is geneated
 
Inside your pipeline you will have allure-results folder with some test data. With `allure generate` command you can generate report based on that data. To encapsulate allure dependencies and this command you can utilize ready-to-use github action from marketplace for that (mentioned above, available in example)

**Reply by SaadUsmanitk** [reply]
hi again, thanks for the great help. I am getting this error on allure history
0s
Run rm -r allure-results/*
  rm -r allure-results/*
  cp -r ./allure-report/history ./allure-results/history
  rm -r allure-report/*
  shell: /usr/bin/bash -e {0}
  env:
    npm_config_cache: /home/runner/.npm
cp: cannot stat './allure-report/history': No such file or directory
Error: Process completed with exit code 1.

history folder is already there

**Reply by Shelex** [reply]
But It tells you that there is no folder `allure-report/history`. Maybe path to folder or current folder is not correct?

**Reply by Shelex** [reply]
First time I have generated allure report manually and committed to main branch.

**Reply by SaadUsmanitk** [reply]
let me do that and i will try again thanks for quick response

**Reply by SaadUsmanitk** [reply]
hi again so i have added the file and everything seems to run at-least on actions but the reports are not generating. https://saadusmanitk.github.io/rsf-qa/allure-report/ this is the data that was generated locally. The reports are not generating on actions.
This is my yaml file. can you check if iam doing something wrong

name: Cypress Run
on: [push]
jobs:
    generate-report:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v2
            - name: Download Cypress
              uses: bahmutov/npm-install@v1
              env:
                  CYPRESS_INSTALL_BINARY: 0

            - name: Check Cypress cache
              run: |
                  npx cypress cache path
                  npx cypress cache list
            - name: restore / cache the binary
              id: cache-cypress
              uses: actions/cache@v1
              with:
                  path: ~/.cache/Cypress
                  key: >-
                      cypress-cache-v2-${{ runner.os }}-${{ hashFiles('**/package.json')
                      }}
            - name: Prepare Allure history
              run: |
                  rm -r allure-results/*
                  cp -r ./allure-report/history ./allure-results/history
                  rm -r allure-report/*
            - name: Install Cypress
              run: |
                  npx cypress install
                  npx cypress cache list
            - name: Cypress run
              uses: cypress-io/github-action@v2
              with:
                browser: chrome
            - name: Generate Allure report
              if: always()
              uses: QuintilianoNery/action-allure-report-V2@0.2.0

            - name: Commit report results
              if: always()
              run: |
                  git config --local user.email "action@github.com"
                  git config --local user.name "GitHub Action"
                  git add allure-report -f
                  git add allure-results -f
                  git commit -m "[chore] autogenerate Allure report" -a

**Comment by thulasibai** [comment]
Hi @Shelex  Do the cypress Allure Plugin supports for the cypress version 13+

**Comment by Shelex** [comment]
Hi @thulasibai 
There is an issue #216 , so I am not sure it works fine after that specific version.
